### PR TITLE
Int64

### PR DESCRIPTION
**Pull request by ebuchman**

### CODE REVIEW

**Comment by odeke-em** [comment]
@ebuchman this change does more than just the int64 usage though, it also uses a different protobuf repo, could we perhaps separate those or at least mention it in the commit info? Sorry to sound like a stickler but I believe it will help things in general to isolate such changes because the protobuf provider change is a big one :)

**Comment by ebuchman** [comment]
this doesn't change the provider. that was done in https://github.com/tendermint/abci/pull/145. Would love another pair of eyes on that one too if you have the chance :)

**Comment by odeke-em** [comment]
Ooh I see, thanks for the reference, let me take a look at it.

**Comment by odeke-em** [comment]
I've given #145 some eyes, looks great, nothing jumped out the page, except just the suggestion to handle the error after invoking `goimports`

File: types/types.proto [hunk 1]
```diff
@@ -178,13 +178,13 @@ message BlockID {
 }
 
 message PartSetHeader {
-  uint64 total = 1;
+  int64 total = 1;
```
Comment (melekes) [inline types/types.proto]: Are we subtracting total somewhere? I thought we just set it once

File: types/types.proto [hunk 2]
```diff
@@ -162,9 +162,9 @@ message ResponseEndBlock{
 
 message Header {
   string chain_id = 1;
-  uint64 height = 2;
-  uint64 time = 3;
-  uint64 num_txs = 4;
+  int64 height = 2;
+  int64 time = 3;
+  int64 num_txs = 4;
```
Comment (melekes) [inline types/types.proto]: same here. or we decided to completely get rid of uint64?

Comment (ebuchman) [reply types/types.proto]: You're probably right, but I think best to just be consistent, no?

**Pull request merged**